### ISSUE TITLE
Add operations with no path and object dot notation not supported

### ISSUE THREAD

**Issue by blairlunceford** [thomaspoignant/scim-patch#132]
**Describe the bug**
Azure AD SCIM sends their add operations for `name` attributes in the following format:

```
{
  "op": "add",
  "value": {
    "name.givenName": "John",
    "name.familyName": "Doe",
    "name.formatted": "John Doe"
  }
}
```

When you apply this operation to a resource using this library, it doesn't recognize the object dot notation, and adds the attributes in the following way:

```
{
  name.givenName: "John",
  name.familyName: "Doe",
  name.formatted: "John Doe"
}
```

However, we would expect them to be added like the following:

```
{
  name: {
    givenName: "John",
    familyName: "Doe",
    formatted: "John Doe"
  }
}
```

**To Reproduce**
```
const scimPatch = require("scim-patch")
const assert = require("assert")

const resource = {
  email: 'jdoe@example.com',
}

const operation = {
  op: 'add',
  value: {
    'name.givenName': 'John',
    'name.familyName': 'Doe',
    'name.formatted': 'John Doe',
  },
}

const expected = {
  email: 'jdoe@example.com',
  name: {
    givenName: "John",
    familyName: "Doe",
    formatted: "John Doe",
  },
}


const patchedResource = scimPatch.scimPatch(resource, [operation])

assert.deepStrictEqual(patchedResource, expected)
```

**Expected behavior**
The library should support parsing of attributes in the object dot notation. This only occurs in one level of nesting, and just for add operations.

**Additional context**
I reached out to Microsoft support because I didn't think this agreed with the SCIM RFC. They quoted the following in their argument that it does agree with the RFC:

> Took me a bit to find it in the spec – but it’s here: https://datatracker.ietf.org/doc/html/rfc7644#page-37
> The following example shows how to add one or more attributes to a User resource without using a "path" attribute.

```
PATCH /Users/2819c223-7f76-453a-919d-413861904646
   Host: example.com
   Accept: application/scim+json
   Content-Type: application/scim+json
   Authorization: Bearer h480djs93hd8
   If-Match: W/"a330bc54f0671c9"
 
   {
     "schemas":
       ["urn:ietf:params:scim:api:messages:2.0:PatchOp"],
     "Operations":[{
       "op":"add",
       "value":{
         "emails":[
           {
             "value":babs@jensen.org,
             "type":"home"
           }
         ],
         "nickname":"Babs"
     }]
   }
 
   In the above example, an additional value is added to the
   multi-valued attribute "emails".  The second attribute, "nickname",
   is added to the User resource.  If the resource already had an
   existing "nickname", the value is replaced per the processing rules
   above for single-valued attributes.
```

> The section I previously linked (page 37 or 7644) combined with https://datatracker.ietf.org/doc/html/rfc7644#section-3.10 together explain how we’re allowed to do this.

```
3.10.  Attribute Notation
 
   All operations share a common scheme for referencing simple and
   complex attributes.  In general, attributes are uniquely identified
   by prefixing the attribute name with its schema URN separated by a
   colon (":") character; e.g., the core User resource attribute
   'userName' is identified as
   "urn:ietf:params:scim:schemas:core:2.0:User:userName".  Clients MAY
   omit core schema attribute URN prefixes but SHOULD fully qualify
   extended attributes with the associated schema extension URN to avoid
   naming conflicts.  For example, the attribute 'age' defined in
   "urn:ietf:params:scim:schemas:exampleCo:2.0:hr" is uniquely
   identified as "urn:ietf:params:scim:schemas:exampleCo:2.0:hr:age".
   Complex attributes' sub-attributes are referenced via nested dot
   ('.') notation, i.e., {urn}:{Attribute name}.{Sub-Attribute name}.
   For example, the fully qualified path for a User's givenName is
   "urn:ietf:params:scim:schemas:core:2.0:User:name.givenName".  All
   facets (URN, attribute, and sub-attribute name) of the fully encoded
   attribute name are case insensitive.
```

**Comment by thomaspoignant** [comment]
Hello @blairlunceford thanks a lot to report that issue.

I was not aware that Azure use this kind of notation and yes this is not currently support by the lib.

I will check this as soon as possible, but in the mean time if you want to have a look your self, feel free to open a PR.

**Comment by thomaspoignant** [comment]
@blairlunceford I fixed the issue, and a new version (`v0.5.2`) is available on npm.
If you still have issues let me know.
